Feature/KAS-3105 Pub v3: Details: Decisions reference documents panel

mdv-professional opened the pull request:
[Jira](https://kanselarij.atlassian.net/browse/KAS-3105)
---
Ik heb wat moeite gehad met dit issue.
1. De functionaliteit was niet volledig omschreven.
2. De DocumentCard moest herwerkt of gekopieerd worden. Ik had moeite met bepalen waar de grens lag.

**Mijn ontwikkelproces**
Ik ben begonnen met een branch waarin ik de `ReferenceDocumentCard` meer DDAU probeerde te maken: [[branch:decisions-ref-docs-2](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/tree/feature/KAS-3105-pub-v3-details--decisions-ref-docs-2)] Dit bleek een hele opgave. Ik ben hiervan afgestapt en in deze branch een kopie gemaakt van de DocumentCard met minimale wijzigingen voor referentiedocumenten. De vraagtekens heb ik hierin onuitgewerkt gelaten.

**¿Vraagtekens?**
- **Referentiedocument toevoegen**
  Er is geen prototype voor deze modal. Ik heb ervoor gekozen om een aantal pagina's toe te voegen. Misschien is een aantal woorden ook zinvol?
- **Nieuwe versie toevoegen**
  - Het kan nuttig zijn hier dezelfde modal te gebruiken als voor "Referentiedocument toevoegen".
     Voordelen:
     - Aantal pagina's kan worden meegegen.
     - Consistente UI
     - AccessLevel is niet afhankelijk van `ReferenceDocumentCard`.
     Dit vereist een meer DDAU aanpak van `ReferenceDocumentCard`. (Gedaan in [[branch:decisions-ref-docs-2](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/tree/feature/KAS-3105-pub-v3-details--decisions-ref-docs-2)])
  - [Besproken met Smooth Sailing] Niet geimplementeerd Bij het toevoegen van een versie zou de vorige versie geen referentiedocument meer mogen zijn. Bij het verwijderen moet de vorige versie terug gemarkeerd worden als referentiedocument.
  
  Persoonlijk denk ik dat versies de zaken hier veel complexer maken en ik zie de meerwaarde er niet van in.
- **AccessLevel**
  - [Besproken met @MikiDi] Niet geimplementeerd: AccessLevel niet-editeerbaar maken. 
  - [Besproken met @MikiDi] Niet uitgeklaard: is dit relevant voor door OVRB beheerde documenten?
    Indien ja => Wat is het correcte access-level voor een door OVRB beheerd document?
- **Vertaling aanvragen**
  - [Besproken met Smooth Sailing] Niet uitgeklaard.

**Code-gewijs**
- Ik zou ervoor kiezen om uiteindelijk dezelfde component te gebruiken onder agenda en publicaties, gezien dit de design-filosofie is van Smooth Sailing. Ik zou daarvoor de huidige `Documents::DocumentCard` meer DDAU maken en deze enerzijds wrappen in een `AgendaDocumentCard` voor de Agenda en anderzijds de juiste @ttributes mee te geven onder `/publications/decisions`. Op die manier kan het template hetzelfde zijn.
- Indien de `ReferenceDocumentCard` een op zichzelf staande component blijft, kunnnen de load...() aangepast/verwijderd worden. Ik zou kiezen om enkel het document in kwestie opnieuw te laden, maar ik weet dat meestal een model-refresh wordt geprefereerd in Kaleidos.
- (minor) Ik zou de component `Utils::Documents::List` verplaatsen. Ik heb deze zelf aangemaakt, maar ze voelt niet aan als een util. Misschien onder de `/documents` iets als `DocumentListCompact`?

**Te doen (reminder voor @mdv-professional)**
- Indien gewenst: validatie van aantal pagina's

**Bugs**
- Wanneer een AccessLevel gewijzigd wordt, update het AccessLevel onder "Versies tonen" niet automatisch.

**Notes voor @brenner-company**
- AccessLevel-dropdown onder "Toon Versies" ( `WebComponents::VlDocument`) wordt te smal.
- Aangepaste loader css staat niet meer over de content om ze te verbergen maar eronder.
- (minor) knoppen in `WebComponents::VlDocument` staan meteen na de documentnaam. Overal elders in de applicatie staan de knoppen rechts uitgelijnd.
- (minor) In `Utils::Documents::List` (vb. in modal "Vertaling aanvragen") staat het document-icon niet helemaal verticaal gecentreerd.
- (minor) De empty-message onder `/publications/overview` is in overeenstemming met het usertest-prototype, maar ik zou hier toch marge verwachten.

Commit:
(copy) DocumentCard
KAS-3105 pub: v3: details: decisions refdocs

Commit:
(move) DecisionsInfoPanel
KAS-3105 pub: v3: details: decisions refdocs

Commit:
add ReferenceDocumentsPanel
KAS-3105 pub: v3: details: decisions refdocs

Commit:
(copy) VlDocument > ReferenceDocumentVersionRow
KAS-3105 pub: v3: details: decisions refdocs

Commit:
update version row component to work with isEditor @ttribute
reason: enable delete for documents not passing via council of ministers
KAS-3105 pub: v3: details: decisions refdocs

Commit:
add missing ReferenceDocumentUploadModal title translation
KAS-3105 pub: v3: details: decisions refdocs

Commit:
add/remove translations for ReferenceDocumentsPanel
KAS-3105 pub: v3: details: decisions refdocs

Commit:
enable edit piece name for OVRB
KAS-3105 pub: v3: details: decisions refdocs

Commit:
remove unused fields of ReferenceDocumentsPanel
KAS-3105 pub: v3: details: decisions refdocs

Commit:
use publicationService.saveTranslationRequest under /publications/publication/translations
KAS-3105 pub: v3: details: decisions refdocs

Commit:
(deprecation-fix) store - /publications/publication/translations
KAS-3105 pub: v3: details: decisions refdocs

Commit:
(deprecation-fix) transitionToRoute - /publications/publication/translations
KAS-3105 pub: v3: details: decisions refdocs

Commit:
use @publicationFlow argument
KAS-3105 pub: v3: details: decisions refdocs

Commit:
(lint-fix)
KAS-3105 pub: v3: details: decisions refdocs

Commit:
(translation)
KAS-3105 pub: v3: details: decisions refdocs

Commit:
(name) consistent params name in ReferenceDocumentsPanel
KAS-3105 pub: v3: details: decisions refdocs

Commit:
(lint-fix) / /publications/publication/translation/documents
KAS-3105 pub: v3: details: decisions refdocs

brenner-company commented:
@mdv-professional 
- Ivm loader: om duidelijk te zijn, dit gaat over de `loadable-content` he? Op het eerste zicht lijkt dat lokaal (op `dev` branch) correct te werken. Kan je anders wat meer info mee geven zoals exacte locatie binnen een pagina of een screenshot?
- Ivm resterende punten: die neem ik op van zodra ik daar tijd voor kan vinden 👍

Commit:
add title to ReferenceDocumentsPanel
KAS-3105 pub: v3: details: decisions refdocs

ghost commented:
@brenner-company OK, sorry, dit is inderdaad opgelost.

Commit:
remove "Vertaling aanvragen" from ReferenceDocumentsPanel
KAS-3105 pub: v3: details: decisions refdocs

erikap committed:
Merge branch 'development' into feature/KAS-3105-pub-v3-details--decisions-ref-docs--copy

erikap committed:
Remove unused return value from updateStatus on publication-service

erikap committed:
Syntax fix

erikap committed:
Add simplified document card for publications

OVRB users don't need all features of the regular document
card. Adding a simplified version of the document card to be used for
publication documents.

erikap committed:
Use publication document card on publication decisions tab

Remove complex logic to add new versions. OVRB users only add new
documents and don't care about versions.

erikap committed:
Split info-panel and documents on publication decision tab to separate routes

erikap committed:
Remove page count from reference document file upload

Since total number of pages will be manually added on the
publication-flow there is no need to capture the number of pages per
document since the sum will not be automatically calculated.

erikap committed:
Add delete button on publication document card

erikap committed:
Remove refactored reference document components

erikap committed:
Revert changes on publication translations documents controller

Since request translation isn't part of this feature-branch anymore,
rollback changes that may cause conflicts  with other branches

erikap committed:
Merge branch 'development' into feature/KAS-3105-pub-v3-details--decisions-ref-docs--copy

erikap commented:
Het Jira ticket voor deze functionaliteit was inderdaad zeer vaag en niet voldoende uitgewerkt.

Na overleg met @MikiDi hebben we een alternatief pad gekozen voor de document-cards bij publicaties. Aangezien de gebruikers van OVRB geen boodschap hebben aan document-versies, toegangsniveaus, vertrouwelijkheid, etc. hebben we een vereenvoudigde versie gemaakt van de document-card voor publicaties: `Publications::Documents::DocumentCardStep`. Een lijst kan weergegeven worden met  `Publications::Documents::DocumentListSteps`. De vereenvoudigde card toont enkel de document-naam, type en buttons om te bekijken/downloaden.

De enige actie die (optioneel) mogelijk is op de nieuwe card is delete. Door het vereenvoudigen van de document-card is heel wat complexiteit uit de initiele PR verwijderd. 

@brenner-company Ik heb in deze PR CSS uit de usertest branch overgenomen in `app/styles/custom-components/_vlc-procedure-step.scss`. Ik weet niet of dat de juiste werkwijze is?

brenner-company commented:
@erikap De CSS mag momenteel zo overgenomen worden (ik kijk later wel nog of er daar nog opkuis werk nodig is). Enkel de locatie mag misschien aangepast worden: `app/styles/custom-components/_vlc-procedure-step.scss` ➡️ `app/styles/au-kaleidos-css/_auk-additions.scss` (is van belang voor CSS specificity).

erikap committed:
Merge branch 'development' into feature/KAS-3105-pub-v3-details--decisions-ref-docs--copy

erikap committed:
Move custom CSS from _vlc-procedure-step.scss to _auk-additions.scss

As suggested by @brenner-company

erikap committed:
Fix publication status icon in overview table

MikiDi committed:
Add CSS comment clarifying usage context of vlc-procedure-step

brenner-company commented:
@mdv-professional Om even op te volgen:
- Ivm AccessLevel-dropdown onder "Toon Versies": ik zie zelf lokaal niet direct een issue, maar als in de toekomst een voorbeeld zou tegen komen van een foute weergave, stuur dan zeker een screenshot door.
- Ivm de resterende item: deze neem ik morgennamiddag even op 👍
